sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q32.sf100/explain.txt

```diff
@@ -1,35 +1,37 @@
 == Physical Plan ==
-CollectLimit (31)
-+- * Project (30)
-   +- * BroadcastHashJoin Inner BuildRight (29)
-      :- * Project (27)
-      :  +- * BroadcastHashJoin Inner BuildLeft (26)
-      :     :- BroadcastExchange (22)
-      :     :  +- * Project (21)
-      :     :     +- * BroadcastHashJoin Inner BuildLeft (20)
-      :     :        :- BroadcastExchange (5)
-      :     :        :  +- * Project (4)
-      :     :        :     +- * Filter (3)
-      :     :        :        +- * ColumnarToRow (2)
-      :     :        :           +- Scan parquet default.item (1)
-      :     :        +- * Filter (19)
-      :     :           +- * HashAggregate (18)
-      :     :              +- Exchange (17)
-      :     :                 +- * HashAggregate (16)
-      :     :                    +- * Project (15)
-      :     :                       +- * BroadcastHashJoin Inner BuildRight (14)
-      :     :                          :- * Filter (8)
-      :     :                          :  +- * ColumnarToRow (7)
-      :     :                          :     +- Scan parquet default.catalog_sales (6)
-      :     :                          +- BroadcastExchange (13)
-      :     :                             +- * Project (12)
-      :     :                                +- * Filter (11)
-      :     :                                   +- * ColumnarToRow (10)
-      :     :                                      +- Scan parquet default.date_dim (9)
-      :     +- * Filter (25)
-      :        +- * ColumnarToRow (24)
-      :           +- Scan parquet default.catalog_sales (23)
-      +- ReusedExchange (28)
+CollectLimit (33)
++- * Project (32)
+   +- * LocalLimit (31)
+      +- * Project (30)
+         +- * BroadcastHashJoin Inner BuildRight (29)
```

Comment:
@c21 @maropu Push down `LocalLimit` through `Project` can not benefit this case.

Reply:
I am wondering why `LocalLimit` does not push through second project? (`Project (30)`)

Reply:
Its child is Inner join and join condition is not empty. We can not pushdown limit through it.

Reply:
> Its child is Inner join and join condition is not empty. We can not pushdown limit through it.

I understand that. My question is why `LocalLimit` is in the middle of two `Project`? Shouldn't it be 

```
Project
- Project
  - LocalLimit
    - BroadcastHashJoin
```

I might miss something.

Reply:
It removed by `ColumnPruning `:
```
=== Applying Rule org.apache.spark.sql.catalyst.optimizer.ColumnPruning ===
 GlobalLimit 10                                                      GlobalLimit 10
 +- LocalLimit 10                                                    +- LocalLimit 10
    +- Project [1 AS 1#10]                                              +- Project [1 AS 1#10]
       +- LocalLimit 10                                                    +- LocalLimit 10
          +- Project                                                          +- Project
             +- LocalLimit 10                                                    +- LocalLimit 10
!               +- Join Inner, (a#6L = a#8L)                                        +- Project
!                  :- Project [a#6L]                                                   +- Join Inner, (a#6L = a#8L)
!                  :  +- Filter isnotnull(a#6L)                                           :- Project [a#6L]
!                  :     +- Relation default.t1[a#6L,b#7L] parquet                        :  +- Filter isnotnull(a#6L)
!                  +- Project [a#8L]                                                      :     +- Relation default.t1[a#6L,b#7L] parquet
!                     +- Filter isnotnull(a#8L)                                           +- Project [a#8L]
!                        +- Relation default.t2[a#8L,b#9L] parquet                           +- Filter isnotnull(a#8L)
!                                                                                               +- Relation default.t2[a#8L,b#9L] parquet
 
```